From 2836c524dce1b4ca480a847a7db7efe90e378ded Mon Sep 17 00:00:00 2001
From: Jiang Xin <zhiyou.jx@alibaba-inc.com>
Date: Fri, 2 Sep 2022 15:40:17 +0800
Subject: [PATCH 3/4] FIXME: dissble failed testcase in t7300 on Windows

Signed-off-by: Jiang Xin <zhiyou.jx@alibaba-inc.com>
---
 t/t7300-clean.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/t/t7300-clean.sh b/t/t7300-clean.sh
index 0399701e62..2116ff4fc3 100755
--- a/t/t7300-clean.sh
+++ b/t/t7300-clean.sh
@@ -725,7 +725,7 @@ test_expect_success 'git clean works if a glob is passed with -d' '
 	test_path_is_missing d2/ut
 '
 
-test_expect_success MINGW 'handle clean & core.longpaths = false nicely' '
+test_expect_failure MINGW 'handle clean & core.longpaths = false nicely' '
 	test_config core.longpaths false &&
 	a50=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa &&
 	mkdir -p $a50$a50/$a50$a50/$a50$a50 &&
-- 
patchwork
